### ISSUE TITLE
[MaterialXView] Failed to generate albedo table shader

### ISSUE THREAD

**Issue by avseoul** [materialx/MaterialX#800]
Hi, not sure this is the correct channel that I could post for help. Please let me know if there is more proper channel for these type of issues, like user forum or so. 

I'm having an issue that it fails to generate the albedo table shader. 
![Screen Shot 2021-12-23 at 1 28 35 PM](https://user-images.githubusercontent.com/3428464/147279193-0654fc70-8ac4-48a5-aa2a-4e7d35456f4c.png)

This is some logs I got from the app:
```
Warning: Debuggee TargetArchitecture not detected, assuming x86_64.
=thread-selected,id="1"
=library-unloaded,id="/Users/avseoul/Development/MaterialX/build/source/MaterialXView/MaterialXView",target-name="/Users/avseoul/Development/MaterialX/build/source/MaterialXView/MaterialXView",host-name="/Users/avseoul/Development/MaterialX/build/source/MaterialXView/MaterialXView"

...

@"2021-12-23 12:27:25.455706-0500 MaterialXView[14474:177267] SecTaskLoadEntitlements failed error=22 cs_flags=20, pid=14474\r\n"
@"2021-12-23 12:27:25.455798-0500 MaterialXView[14474:177267] SecTaskCopyDebugDescription: MaterialXView[14474]/0#-1 LF=0\r\n"

...

@"2021-12-23 12:27:26.147544-0500 MaterialXView[14474:177267] [General] ERROR: Setting <GLFWContentView: 0x102e883b0> as the first responder for window <GLFWWindow: 0x102f16b90>, but it is in a different window ((null))! This would eventually crash when the view is freed. The first responder will be set to nil.\r\r\n"
@"(\r\n"
@"\t0   AppKit                              0x00007fff22d849e1 -[NSWindow _validateFirstResponder:] + 449\r\n"
@"\t1   AppKit                              0x00007fff22d847dc -[NSWindow _setFirstResponder:] + 31\r\n"
@"\t2   AppKit                              0x00007fff22ea8b79 -[NSWindow _realMakeFirstResponder:] + 308\r\n"
@"\t3   MaterialXView                       0x0000000100443a4d createWindow + 957\r\n"
@"\t4   MaterialXView                       0x000000010044348f _glfwPlatformCreateWindow + 63\r\n"
@"\t5   MaterialXView                       0x000000010043c546 glfwCreateWindow + 790\r\n"
@"\t6   MaterialXView                       0x0000000100339ef0 _ZN7nanogui6ScreenC2ERKN5Eigen6MatrixIiLi2ELi1ELi0ELi2ELi1EEERKNSt3__112basic_stringIcNS6_11char_traitsIcEENS6_9allocatorIcEEEEbbiiiiijj + 992\r\n"
@"\t7   MaterialXView                       0x0000000100089132 _ZN6ViewerC2ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_S8_RKN17MaterialX_v1_38_314FileSearchPathERKNS0_6vectorINS9_8FilePathENS4_ISE_EEEEiiRKNS9_6Color3Ei + 194\r\n"
@"\t8   MaterialXView                       0x000000010008a65a _ZN6ViewerC1ERKNSt3__112basic_stringIcNS0_11char_traitsIcEENS0_9allocatorIcEEEES8_S8_RKN17MaterialX_v1_38_314FileSearchPathERKNS0_6vectorINS9_8FilePathENS4_ISE_EEEEiiRKNS9_6Color3Ei + 106\r\n"
@"\t9   MaterialXView                       0x0000000100059bdf main + 6495\r\n"
@"\t10  libdyld.dylib                       0x00007fff2048af3d start + 1\r\n"
@")\r\n"

...

The program '/Users/avseoul/Development/MaterialX/build/source/MaterialXView/MaterialXView' has exited with code 0 (0x00000000).
```

If I skip `Viewer.updateAlbedoTable()` in `Viewer.renderFrame()` by commenting out then it at least seems to work.
![Screen Shot 2021-12-23 at 1 26 39 PM](https://user-images.githubusercontent.com/3428464/147279216-398bf68f-2eb3-4e5a-ae95-1e830012eccb.png)

I'm building the viewer from [7a088af](https://github.com/materialx/MaterialX/commit/7a088afe215212341fccab3b78ff2435c21d6a67) where the `v1.38.3` is tagged, on Intel macOS Big Sur 11.6

Any help would be appreciated!

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @avseoul, and it's not an issue that I've encountered before.  I'd be interested in thoughts from other macOS developers, since I personally only have access to builds of MaterialXView on Windows and Linux.

**Comment by avseoul** [comment]
Thanks for following it up @jstone-lucasfilm 

I was able to get a little more information from the issue. Somehow the error logs weren't printed on the alert window so I had to print them manually. It seems the issue is coming from the fragment shader of the `albedoTableShader`. The logs are following: 

```
GLSL program creation error: 
  Error in compiling fragment shader:
    ERROR: 0:158: '' : syntax error: incorrect preprocessor directive
    ERROR: 0:158: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
    ERROR: 0:201: '' : syntax error: incorrect preprocessor directive
    ERROR: 0:201: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
    ERROR: 0:311: '' : syntax error: incorrect preprocessor directive
    ERROR: 0:311: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
    ERROR: 0:361: '' : syntax error: incorrect preprocessor directive
    ERROR: 0:361: '' : syntax error: unexpected tokens following #if preprocessor directive - expected a newline
```

I might have been missing something but it seems the shader is assembled in runtime so I wasn't sure where I should look into. Any help would be appreciated!

**Comment by jstone-lucasfilm** [comment]
Do you get the same error messages when you run the _prebuilt binaries_ for MaterialX 1,38.3 on macOS Big Sur?

https://github.com/materialx/MaterialX#pre-built-binaries

That should help us to determine whether it's a local build issue or a broader compatibility issue with macOS Big Sur.  Let us know what you find!

**Comment by avseoul** [comment]
Yes, I'm getting the same issue when I run the prebuilt binaries. 

Just to make sure that I'm doing it correct way, how I did is that I run the executable in the bin folder from the terminal. Then I see this window same as I saw from the manual build:

![a](https://user-images.githubusercontent.com/3428464/148247630-ffce7073-e0f9-4885-b3a3-a92616430bcf.png)

Let me know if there is anything you want me to test from my end!

**Comment by jstone-lucasfilm** [comment]
Thanks for that additional data, @avseoul, and it does sound like we have an uncaught shader compilation issue on macOS Big Sur.  I'm CC'ing @ashwinbhat, who has greater expertise with MaterialX on that platform, to get his initial thoughts.

**Comment by avseoul** [comment]
Awesome, thanks for looking into it! Please let me know if you need any test from my end.

**Comment by kwokcb** [comment]
That would be very useful. Can you dump out the shader (press G key in the viewer) and the associated materialx file?
 Looks like the `#define` statements are bad for whatever reason.

**Comment by avseoul** [comment]
[standard_surface_default.zip](https://github.com/materialx/MaterialX/files/7816795/standard_surface_default.zip)
Np! Attached the shaders and .mtlx. This is what I see when I press G, just to make sure if I'm sharing the correct shaders. 
![Screen Shot 2022-01-05 at 12 52 51 PM](https://user-images.githubusercontent.com/3428464/148265130-74a4b20d-c978-4fdf-bc0e-a5ade19d8275.png)

**Comment by ashwinbhat** [comment]
@avseoul I've created a PR to address this issue. Please let us know if there any issue. 
Thanks

**Comment by avseoul** [comment]
@ashwinbhat I've just tried the branch and looking great! The issue seems to be gone. Thanks so much!
![Screen Shot 2022-01-05 at 7 17 49 PM](https://user-images.githubusercontent.com/3428464/148308323-7cd24387-de46-48f6-9de5-c44b1711cc89.png)

**Comment by jstone-lucasfilm** [comment]
Fixed in pull request #807.  Thanks to @avseoul for this report, and to @ashwinbhat for the fix!
